charts/tidb-cluster/templates/scripts/_start_pd.sh.tpl

```diff
@@ -30,7 +30,7 @@ fi
 
 # the general form of variable PEER_SERVICE_NAME is: "<clusterName>-pd-peer"
 cluster_name=`echo ${PEER_SERVICE_NAME} | sed 's/-pd-peer//'`
-domain="${HOSTNAME}.${PEER_SERVICE_NAME}.${NAMESPACE}.svc"
+domain="${POD_NAME}.${PEER_SERVICE_NAME}.${NAMESPACE}.svc"
```

Comment:
We should use `POD_NAME` instead of `HOSTNAME` environment. In Kubernetes, the hostname in the pod is implicitly set to the host hostname when the pod is running in host network (xref: https://github.com/kubernetes/kubernetes/issues/12893). Currently, there is no way to change this behavior.

Reply:
`POD_NAME` is undefined if using old tidb-operator. To be compatible with old tidb-operator, set `POD_NAME` to `$(hostname)` or `HOSTNAME` if `POD_NAME` is undefined.